### PR TITLE
MERC-991 & MERC-992 custom templates lookup fix

### PR DESCRIPTION
**Pull request by mcampa**
Fixes:
- Stencil CLI should support page names with and without the trailing slash
- Incorrect name of custom directory

@sherrybc

### CODE REVIEW

**Comment by mcampa** [comment]
Needs testing before merging to master @bc-ranji

File: server/plugins/renderer/renderer.module.js [hunk 1]
```diff
@@ -332,69 +334,30 @@ internals.redirect = function (response, request, callback) {
...
-internals.getTemplatePath = function (request, defaultTemplateFile) {
-    var customLayouts = internals.options.customLayouts;
-    var templatePath = internals.findDeepTemplate(customLayouts, request.path);
+internals.getTemplatePath = function (path, data) {
+    var customLayouts = internals.options.customLayouts || {};
+    var pageType = data.page_type;
+    var templatePath;
+
+    if (internals.validCustomTemplatePageTypes.indexOf(pageType) && _.isPlainObject(customLayouts[pageType])) {
+        templatePath = _.findKey(customLayouts[pageType], function(p) {
```
Comment (sherrybc) [inline server/plugins/renderer/renderer.module.js]: I haven't dig into this yet so I am not sure what `customLayouts` return, if `customLayouts[pageType]` contains more than object, are you aware that ``` `_.findkey` returns the key of the first element predicate returns truthy for```

Comment (mcampa) [inline server/plugins/renderer/renderer.module.js]: `customLayouts` has the same value it has in your `.stencil` file:
```
    customLayouts =  {
    "products": {},
    "search": {},
    "brands": {},
    "categories": {},
    "page": {
      "custom.html": "/home/",
      "inspire-me-details.html": "/inspire-me-details/",
      "inspire-me.html": "/inspire-me/",
      "store.html": "/store/",
      "stores.html": "/stores/"
    }
  };
    path = '/home';

templatePath = _.findKey(customLayouts['page'], func)
//  custom.html
```

**Comment by bc-ranji** [comment]
Hi Mario 

Found a bug with brand custom templates not rendering, will show you the issue

Thanks
Ranjitha

**Comment by sherrybc** [comment]
👍

**Comment by bc-ranji** [comment]
💚

**Pull request merged**